scripts/generate-changelog.mjs

```diff
@@ -0,0 +1,186 @@
+ *   $ node ./scripts/generate-changelog.mjs 10961 typescript
+ */
+
+import fs from "node:fs/promises";
+import path from "node:path";
+import fetch from "node-fetch";
+import createEsmUtils from "esm-utils";
+import { CHANGELOG_CATEGORIES } from "./utils/changelog-categories.mjs";
+
+const { __dirname } = createEsmUtils(import.meta);
+
+const [prNumberString, category] = process.argv.slice(2);
+if (!prNumberString || !category) {
+  throw new Error("Two args are required.");
+}
+const prNumber = convertToNumber(prNumberString);
+assertCategory(category);
+
+const { title, user } = await getPr(prNumber);
+
+const newChangelog = await createChangelog(title, user, prNumber, category);
+
+await addNewChangelog(prNumber, category, newChangelog);
+
+console.log("Done");
+
+/**
+ * @param {number} prNumber
+ * @returns {Promise<{ title: string; user: string }>}
+ */
+async function getPr(prNumber) {
+  // https://docs.github.com/en/rest/reference/pulls#get-a-pull-request
+  const url = `https://api.github.com/repos/prettier/prettier/pulls/${prNumber}`;
+  const response = await fetch(url, {
+    Headers: {
+      "Content-Type": "application/json",
+      Accept: "application/vnd.github.v3+json",
+    },
+  });
+  if (!response.ok) {
+    if (response.status === 404) {
+      throw new Error(`Pull Request #${prNumber} not found.`);
+    }
+    throw new Error(response);
+  }
+  const { title, user } = await response.json();
+  return {
+    title,
+    user: user.login,
+  };
+}
+
+/**
+ * @param {number} prNumber
+ * @param {string} category
+ * @param {string} newChangelog
+ * @returns {Promise<void>}
+ */
+async function addNewChangelog(prNumber, category, newChangelog) {
+  const newChangelogPath = path.resolve(
+    __dirname,
+    `../changelog_unreleased/${category}/${prNumber}.md`
+  );
+  await fs.writeFile(newChangelogPath, newChangelog);
+}
+
+/**
+ * @param {string} title
+ * @param {string} user
+ * @param {number} prNumber
+ * @param {string} string
+ * @returns {string}
+ */
+async function createChangelog(title, user, prNumber, category) {
+  const changelogTemplatePath = path.resolve(
+    __dirname,
+    "../changelog_unreleased/TEMPLATE.md"
+  );
+  const changelogTemplate = await fs.readFile(changelogTemplatePath, "utf-8");
+
+  const titlePart = "Title";
+  const prNumberPart = "#XXXX";
+  const userPart = "@user";
+  const codeBlockPart = "```jsx\n";
+  const inputCommentPart = "// Input\n";
+  const stableCommentPart = "// Prettier stable\n";
+  const mainCommentPart = "// Prettier main\n";
+
+  const syntax = getSyntaxFromCategory(category);
+
+  return changelogTemplate
+    .replace(titlePart, title)
+    .replace(prNumberPart, `#${prNumber}`)
+    .replace(userPart, `@${user}`)
+    .replace(codeBlockPart, `\`\`\`${syntax}\n`)
+    .replace(inputCommentPart, getCommentForSyntax(syntax, "Input") + "\n")
+    .replace(
+      stableCommentPart,
+      getCommentForSyntax(syntax, "Prettier stable") + "\n"
```

Comment:
Do you think we can also auto generate stable and main result by `require('prettier').format` and `require('.').format` ?

Reply:
I think it's possible, but it's hard to think of a good interface. I think at least it isn't in scope of this PR.